pkg/controllers/node/drift.go

```diff
@@ -39,17 +40,36 @@ func (d *Drift) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner
 	if _, ok := node.Annotations[v1alpha5.VoluntaryDisruptionAnnotationKey]; ok {
 		return reconcile.Result{}, nil
 	}
-
+	machineName, ok := node.Labels[v1alpha5.MachineNameLabelKey]
+	if !ok {
+		return reconcile.Result{}, nil
+	}
+	machine := &v1alpha5.Machine{}
+	if err := d.kubeClient.Get(ctx, types.NamespacedName{Name: machineName}, machine); err != nil {
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+	if !machine.StatusConditions().GetCondition(v1alpha5.MachineCreated).IsTrue() {
+		return reconcile.Result{Requeue: true}, nil
+	}
 	// TODO: Add Provisioner Drift
-	drifted, err := d.cloudProvider.IsMachineDrifted(ctx, machine.NewFromNode(node))
+	drifted, err := d.cloudProvider.IsMachineDrifted(ctx, machine)
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("getting drift for node, %w", err)
 	}
-	if drifted {
-		node.Annotations = lo.Assign(node.Annotations, map[string]string{
-			v1alpha5.VoluntaryDisruptionAnnotationKey: v1alpha5.VoluntaryDisruptionDriftedAnnotationValue,
-		})
+	if !drifted {
+		// Requeue after 5 minutes for the cache TTL
+		return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
+	}
+	node.Annotations = lo.Assign(node.Annotations, map[string]string{
+		v1alpha5.VoluntaryDisruptionAnnotationKey: v1alpha5.VoluntaryDisruptionDriftedAnnotationValue,
+	})
+	if err = d.kubeClient.Update(ctx, node); err != nil {
```

Comment:
I thought we don't need to do this since we patch the node in the larger node controller. Did we change this?

Reply:
It's been changed since then